### ISSUE TITLE
Search in specified discrete combinatory space

### ISSUE THREAD

**Issue by hyiltiz** [lacerbi/bads#6]
Is it possible to specify discrete valid values for each dimension of the parameter (of the cost function) such that only those values on the grid are evaluated? A special case would be optimizing a single integer variable. An extension is being able to specify some dimensions of the parameter to be discrete and specify the discrete space over which it takes values. Some problems are discrete in nature, but combinatorics kills any attempt to do greedy grid search due to the curse of dimensionality.

**Comment by lacerbi** [comment]
Yep, that's the solution that I currently recommend as a quick fix, but it's clearly not ideal.
I had this request before and I have been thinking of implementing integer variables, but not happening soon, it does require some tinkering.
